carbon-route.html

```diff
@@ -223,16 +223,16 @@
     __resetProperties: function() {
       this._setActive(false);
       this._matched = null;
-      //this.tail = { path: null, prefix: null, queryParams: null };
-      //this.data = {};
+      this.tail = { path: null, prefix: null, __queryParams: null };
+      this.data = {};
```

Comment:
Are these changes necessary for this PR to work?

The reason why we've been leaning towards not clearing out `tail` and `data` when a route no longer matches is that we want to do as little work as possible on views that aren't visible.

Consider the scenario where we've got three pages: `/user` `/user/:username/profile` and `/watch/:videoId` with `<user-page>` and `<watch-page>` as top level siblings, and `<user-page>` using `tail` to delegate to a `<user-profile-page>`.

When navigating from `/user/alice/profile` to `/watch/panda-sneeze` we have to do a certain amount of work to set up the panda-sneeze video. From a performance standpoint, it would be nice if we could do the minimum of changes inside `<user-profile-page>` at the same time, as it's invisible to the user at that point.